src/dma.md

```diff
@@ -0,0 +1,348 @@
+# Direct Memory Access (DMA)
+
+This section covers the core requirements for building a memory safe API around
+DMA transfers.
+
+The DMA peripheral is used to perform memory transfers in parallel to the work
+of the processor (the execution of the main program). A DMA transfer is more or
+less equivalent to spawning a thread (see [`thread::spawn`]) to do a `memcpy`.
+We'll use the fork-join model to illustrate the requirements of a memory safe
+API.
+
+[`thread::spawn`]: https://doc.rust-lang.org/std/thread/fn.spawn.html
+
+Consider the following DMA primitives:
```

Comment:
Unless the vendor has done a bad job at designing their DMA peripheral, shared registers should only need read OR (stateless) write operations on them so that no additional synchronization is needed. Is there some problematic case you have in mind?